pydicom/tests/test_encoders_pydicom.py

```diff
@@ -0,0 +1,450 @@
+        assert 1 == ds.SamplesPerPixel
+
+        kwargs = RLELosslessEncoder.kwargs_from_ds(ds)
+        encoded = _encode_frame(ds.PixelData, **kwargs)
+        decoded = _rle_decode_frame(
+            encoded, ds.Rows, ds.Columns, ds.SamplesPerPixel, ds.BitsAllocated
+        )
+        arr = np.frombuffer(decoded, '<u2')
+        arr = reshape_pixel_array(ds, arr)
+
+        assert np.array_equal(ref, arr)
+
+    def test_cycle_16bit_3sample(self):
+        """Test an encode/decode cycle for 16-bit 3 sample/pixel."""
+        ds = dcmread(EXPL_16_3_1F)
+        ref = ds.pixel_array
+        assert ds.BitsAllocated == 16
+        assert ds.SamplesPerPixel == 3
+        assert ds.PixelRepresentation == 0
+
+        kwargs = RLELosslessEncoder.kwargs_from_ds(ds)
+        encoded = _encode_frame(ds.PixelData, **kwargs)
+        decoded = _rle_decode_frame(
+            encoded, ds.Rows, ds.Columns, ds.SamplesPerPixel, ds.BitsAllocated
+        )
+        # The decoded data is planar configuration 1
+        ds.PlanarConfiguration = 1
+        arr = np.frombuffer(decoded, '<u2')
+        arr = reshape_pixel_array(ds, arr)
+
+        assert np.array_equal(ref, arr)
+
+    def test_cycle_32bit_1sample(self):
+        """Test an encode/decode cycle for 32-bit 1 sample/pixel."""
+        ds = dcmread(EXPL_32_1_1F)
+        ref = ds.pixel_array
+        assert 32 == ds.BitsAllocated
+        assert 1 == ds.SamplesPerPixel
+
+        kwargs = RLELosslessEncoder.kwargs_from_ds(ds)
+        encoded = _encode_frame(ds.PixelData, **kwargs)
+        decoded = _rle_decode_frame(
+            encoded, ds.Rows, ds.Columns, ds.SamplesPerPixel, ds.BitsAllocated
+        )
+        arr = np.frombuffer(decoded, '<u4')
+        arr = reshape_pixel_array(ds, arr)
+
+        assert np.array_equal(ref, arr)
+
+    def test_cycle_32bit_3sample(self):
+        """Test an encode/decode cycle for 32-bit 3 sample/pixel."""
+        ds = dcmread(EXPL_32_3_1F)
+        ref = ds.pixel_array
+        assert ds.BitsAllocated == 32
+        assert ds.SamplesPerPixel == 3
+        assert ds.PixelRepresentation == 0
+
+        kwargs = RLELosslessEncoder.kwargs_from_ds(ds)
+        encoded = _encode_frame(ds.PixelData, **kwargs)
+        decoded = _rle_decode_frame(
+            encoded, ds.Rows, ds.Columns, ds.SamplesPerPixel, ds.BitsAllocated
+        )
+        # The decoded data is planar configuration 1
+        ds.PlanarConfiguration = 1
+        arr = np.frombuffer(decoded, '<u4')
+        arr = reshape_pixel_array(ds, arr)
+
+        assert np.array_equal(ref, arr)
+
+    def test_16_segments_raises(self):
+        """Test that trying to encode 16-segments raises exception."""
+        msg = (
+            r"Unable to encode as the DICOM standard only allows "
+            r"a maximum of 15 segments in RLE encoded data"
+        )
+        with pytest.raises(ValueError, match=msg):
+            _encode_frame(b'', samples_per_pixel=3, bits_allocated=64)
```

Comment:
wouldn't that be 24-segments? Maybe you meant `samples_per_pixle=2`?

Reply:
Oh yes, sorry, adapted an existing test and forgot to rename.

Reply:
Rewrote the test instead